ctapipe/image/time_next_neighbor_cleaning.py

```diff
@@ -0,0 +1,742 @@
+                sum_time = sample_time * n_samples
+
+                self.fill_individual_pixel_rate(geometry, image, sum_time, charge_steps)
+
+    def scaled_combfactor_event_display(self):
+        """
+        Hardcoded values from EventDisplay scaled from 2400 pixels to the
+        number of pixels. Nobody really understands where these numbers come
+        from. Should use `combfactor_from_geometry` instead.
+
+        Returns
+        -------
+        comb_factor: ndarray
+            scales combinatorial factor
+        """
+        comb_factor = {
+            "4nn": 12e5,
+            "2+1": 95e4,
+            "3nn": 13e4,
+            "2nn": 6e3,
+            "bound": 2,
+        }  # for 2400 pixels
+
+        comb_factor.update(
+            (x, y * self._geometry.n_pixels / 2400) for (x, y) in comb_factor.items()
+        )
+
+        return comb_factor[self._neighbor_group]
+
+    def combfactor_from_geometry(
+        self, neighbor_group=None, camera_name=None, calculate=False
+    ):
+        """
+        Get the combinatorial factors from the geometry of the camera by
+        counting the number of possible combinations for a given group of
+        pixels.
+        If `calculate == True`, the number of combinations will be calculated
+        for the specified camera and neighbor group. Otherwise the values are
+        read from a lookup table which was filled for complete cameras, e.g.
+        assuming no broken pixels.
+
+        Parameters
+        ----------
+        camera_name: string, None
+        neighbor_group: string, None
+        calculate: bool
+            If true, the value will be calculated from the geometry of the
+            camera. Otherwise the values from a pre filled lookup table are
+            returned
+
+        Returns‚
+        -------
+        comb_factor: int
+            number of possible combinations for camera type and neighbor group
+
+        """
+        # Allow to specify camera and neighbor group by hand. If not set, use
+        # currently selected values.
+        camera_name = camera_name if camera_name else self._camera_name
+        neighbor_group = neighbor_group if neighbor_group else self._neighbor_group
+
+        if not calculate:
+            lookup = dict(
+                LSTCam={"4nn": 75302, "3nn": 19269, "2nn": 5394, "2+1": 71712},
+                ASTRICam={"4nn": 41953, "3nn": 13548, "2nn": 4624, "2+1": 70904},
+                DigiCam={"4nn": 51843, "3nn": 13327, "2nn": 3744, "2+1": 49320},
+                FlashCam={"4nn": 71559, "3nn": 18319, "2nn": 5124, "2+1": 68112},
+                NectarCam={"4nn": 75302, "3nn": 19269, "2nn": 5394, "2+1": 71712},
+                CHEC={"4nn": 36289, "3nn": 11720, "2nn": 4000, "2+1": 61432},
+            )
+            comb_factor = lookup[camera_name][neighbor_group]
+
+        else:
+            from ctapipe.instrument import CameraGeometry
+
+            geometry = CameraGeometry.from_name(camera_name)
```

Comment:
CameraGeometry.from_name is only used for testing purposes, and does not guarantee you get the correct camera geometry for the file you are reading.  Rather you should let the user pass in a SubarrayDescription to this class (As we do with other classes that need instrument information). Then you can just access all the cameras from that object. 

See how we do it elsewhere. There are generally loops like:
```py3
with EventSource(eventfile) as source:
     cleaner = TimeNextNeighborCleaning(subarray=source.subarray)
     ...
     for event in source:
          cleaner.clean(event)
```